packages/gatsby/src/redux/nodes.js

```diff
@@ -103,26 +103,20 @@ exports.saveResolvedNodes = async (nodeTypeNames, resolver) => {
   }
 }
 
-const getNodesAndResolvedNodes = typeName => {
+const addResolvedNodes = (typeName, arr) => {
   const { nodesByType, resolvedNodesCache } = store.getState()
-  const nodes = nodesByType.get(typeName)
-  if (nodes) {
-    const resolvedNodes = resolvedNodesCache.get(typeName)
+  const nodes /*: Map<mixed> */ = nodesByType.get(typeName)
```

Comment:
`Map<mixed>` seems to be a wrong type, as `nodesByType.get` returns an array?

Reply:
No, pretty sure it's a `Map`: https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby/src/redux/index.js#L18
Note that `Map` and `Set` have `.forEach` as well, if that's what you were going off on. And in fact, I tried to clarify that it _isn't_ an array by adding this type hint :)0

Reply:
this line is for the `Map` for the `nodesByType.get`:

https://github.com/gatsbyjs/gatsby/blob/006ecd8458a9b83ec0305059bfd0ff057adcac51/packages/gatsby/src/redux/index.js#L22

Reply:
So it contains a map of nodes by id. I see, thanks!